### PR TITLE
Admin users

### PR DESCRIPTION
**Pull request by AlonaDorozhy**
Most bugs fixed
All your comments are accepted, I will work on them in a further correction
I will not continue to change work today, I don't want to make the situation worse for other team members

### CODE REVIEW

File: src/app/admin-users/admin-users.component.scss [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    font-size: 16px;  
+    border-radius: 5px;
+    color: $button-text-color;
+    border: 1px solid $button-green;
+    width: 30%;
+    height: 35px;
+    background-color: $border;
+    margin: auto;
+}
+.buttonDelet{
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.scss]: it is better to use material buttons instead, in any case all this properties are almost identical for all the buttons so it is better to apply all common one in one block (alternatively create a mixing to style the button, so you can pass the color and it does the trick)
But to switch is to material is the best option

File: src/app/admin-users/admin-users.component.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  selector: 'app-admin-users',
+  templateUrl: './admin-users.component.html',
+  styleUrls: ['./admin-users.component.scss']
+})
+export class AdminUsersComponent implements OnInit {
+
+  constructor(private userService: UsersService) { }
+  users;
+  managers;
+  ngOnInit(): void {
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.ts]: properties should go first;
add visibility modifiers;

File: src/app/admin-users/admin-users.component.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    this.userService.getUserInfo().subscribe((res) => {
+      this.users = res;
+      return this.users;
+    });
+
+    this.userService.getManagers().subscribe((res) => {
+      this.managers = res;
+      return this.managers;
+    });
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.ts]: use takeUntil or async pipe

File: src/app/admin-users/admin-users.models.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+export class User {
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.models.ts]: 1 entity per file

File: src/app/admin-users/admin-users.module.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    MatInputModule
+
+  ],
+  exports: [
+    RouterModule,
+    MatTabsModule,
+  ]
+})
+export class AdminUsersModule { }
+
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.module.ts]: remove all extra new lines

File: src/app/admin-users/helpers/helpers.component.html [hunk 6]
```diff
@@ -0,0 +1,53 @@
+<div>
+
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helpers.component.html]: remove all extra new lines and spaces

File: src/app/admin-users/helpers/helpers.component.ts [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+@Component({
+  selector: 'app-helpers',
+  templateUrl: './helpers.component.html',
+  styleUrls: ['./helpers.component.scss', ]
+})
+
+
+export class HelpersComponent implements OnInit, OnDestroy {
+  private unsubscribe: Subject<void> = new Subject();
+  user: User[];
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helpers.component.ts]: add all missing visibility modifiers

File: src/app/admin-users/helpers/helpers.component.ts [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  helperId;
+  public visibleFields = false;
+  public buttonName: any = ' visibleFields';
```
Comment (ksenya-ta) [inline src/app/admin-users/helpers/helpers.component.ts]: why do you need this buttonName property?

File: src/app/admin-users/managers/managers.component.html [hunk 9]
```diff
@@ -0,0 +1,52 @@
+<div>
+    <div *ngIf="manager" class="comment-wrap main-wrap">
+        <div class="author-avatar">
+            <img src="https://icons-for-free.com/iconfiles/png/512/facebook+profile+user+profile+icon-1320184041317225686.png"
+                class=" avatar" alt="Menager Avatar">
+        </div>
+        <div class="comment-body">
+            <form [formGroup]="managerForm" (ngSubmit)="editInfoManager()">
+                <div class="comment-author-wrap ">
+                    <p> name: <span *ngIf="! visibleFields">
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/managers.component.html]: too much unnecessary ngIf checks, just make 2 blocks with the markup for 1 case and the markup for other one

File: src/app/admin-users/users.service.ts [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+apiManager = 'https://cors-anywhere.herokuapp.com/https://familynetserver.azurewebsites.net/api/v1/representatives/';
+
+  constructor(private httpClient: HttpClient) { }
+
+  public getAllUsers(url?: string): Observable<User[]> {
+    return this.httpClient.get<User[]>(`${this.apiUsers}`);
+  }
+  public getManagers(url?: string) {
+    return this.httpClient.get<Manager[]>(`${this.apiManager}`);
+  }
```
Comment (ksenya-ta) [inline src/app/admin-users/users.service.ts]: it is 2 different entities, it should not be part of a single service

File: src/app/app.module.ts [hunk 11]
```diff
@@ -10,9 +10,20 @@ import { LoginModule } from './login/login.module';
 import { RegistrationUserModule } from './registration-user/registration-user.module';
 import { NavbarComponent } from './navbar/navbar.component';
 import { MatButtonModule } from '@angular/material/button';
+<<<<<<< HEAD
```
Comment (ksenya-ta) [inline src/app/app.module.ts]: resolve conflicts properly

File: src/app/navbar/navbar.component.html [hunk 12]
```diff
@@ -6,7 +6,12 @@
   <a mat-button [routerLink]="['/shelters']">
     Find shelter
   </a>
+<<<<<<< HEAD
```
Comment (ksenya-ta) [inline src/app/navbar/navbar.component.html]: same here

File: src/app/admin-users/admin-users.component.ts [hunk 13]
```diff
@@ -0,0 +1,27 @@
...
+})
+export class AdminUsersComponent implements OnInit {
+
+  constructor(private userService: UsersService) { }
+  users;
+  managers;
+  ngOnInit(): void {
+    this.userService.getUserInfo().subscribe((res) => {
+      this.users = res;
+      return this.users;
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.ts]: no need in all this returns

File: src/app/admin-users/admin-users.component.ts [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+  templateUrl: './admin-users.component.html',
+  styleUrls: ['./admin-users.component.scss']
+})
+export class AdminUsersComponent implements OnInit {
+
+  constructor(private userService: UsersService) { }
+  users;
+  managers;
+  ngOnInit(): void {
+    this.userService.getUserInfo().subscribe((res) => {
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.ts]: unsubscribe?

File: src/app/admin-users/admin-users.component.html [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+                <div> <a [routerLink]="['/users', user.id]">
+                        <button class="button">More</button></a>
+                </div>
+            </div>
+        </mat-tab>
+        <mat-tab label="Managers">
+            <div class="side">
+                <button class="buttonAddManager">Add Manager</button>
+            </div>
+            <div *ngFor="let manager of managers " class="comment-wrap main-wrap">
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.html]: and also you better you async pipe here

File: src/app/admin-users/admin-users.component.html [hunk 16]
```diff
@@ -0,0 +1,51 @@
+<div>
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.html]: this file should be split in to 5 separate components:
the component with titles and 2 other components
2 list components
2 items components

File: src/app/admin-users/admin-users.component.ts (hunk 13 above)
Comment (AlonaDorozhy) [reply src/app/admin-users/admin-users.component.ts]: I need them without them I do not receive data to the user

File: src/app/admin-users/admin-users.component.html (hunk 16 above)
Comment (ksenya-ta) [reply src/app/admin-users/admin-users.component.html]: this one can be moved to a separate task as technical debt for future sprint

File: src/app/admin-users/admin-users.component.html [hunk 17]
```diff
@@ -4,36 +4,31 @@
             <div class="side">
                 <button class="buttonAddHelper">Add new Helper </button>
             </div>
-            <div *ngFor="let user of users " class="comment-wrap main-wrap">
+            <div *ngFor="let user of users$ | async; trackBy: userTrackBy" class="comment-wrap main-wrap">
```
Comment (AlonaDorozhy) [inline src/app/admin-users/admin-users.component.html]: added   async and trackBy

File: src/app/admin-users/admin-users.component.scss [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+
+.main-wrap {
+    width: 60%;
+    margin: 25px 0 0 30%;
+}
+.avatar {
+    width: 90%;
+    margin: auto;
+}
+.comment-wrap {
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.scss]: empty line between rules

File: src/app/admin-users/admin-users.component.ts [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+import { User, Manager } from './admin-users.models';
+
+@Component({
+  selector: 'app-admin-users',
+  templateUrl: './admin-users.component.html',
+  styleUrls: ['./admin-users.component.scss']
+})
+
+
+export class AdminUsersComponent implements OnInit {
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.ts]: remove empty line between class and its decorator

File: src/app/admin-users/admin-users.component.ts [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+  public users$: Observable<User[]>;
+  public managers$: Observable<Manager[]>;
+  help: User[];
+
+
+  constructor(private helperService: HelpersService, private managerService: ManagersService) { }
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.ts]: don't add more than 1 empty line

File: src/app/admin-users/admin-users.component.ts [hunk 21]
```diff
@@ -0,0 +1,31 @@
...
+  ngOnInit(): void {
+    this.users$ = this.helperService.getUserInfo();
+    this.managers$ =  this.managerService.getManagers();
+  }
+  usertrackBy(index: number, user: User): string {
```
Comment (ksenya-ta) [inline src/app/admin-users/admin-users.component.ts]: userTrackBy, same below

File: src/app/app.module.ts [hunk 22]
```diff
@@ -21,7 +20,7 @@ import { MatButtonModule } from '@angular/material/button';
     MatButtonModule,
     LoginModule,
     RegistrationUserModule,
-    DonationModule
```
Comment (ksenya-ta) [inline src/app/app.module.ts]: why the donations module was removed?

Comment (YanaSuprun) [reply src/app/app.module.ts]: The same question. It's my module, add it back, pls.

Comment (YanaSuprun) [reply src/app/app.module.ts]: The same question. It's my module, add it back, pls.

Comment (YanaSuprun) [reply src/app/app.module.ts]: The same question. It's my module, add it back, pls.

File: src/app/admin-users/managers/managers.component.ts [hunk 23]
```diff
@@ -0,0 +1,61 @@
...
+  selector: 'app-managers',
+  templateUrl: './managers.component.html',
+  styleUrls: ['./managers.component.scss']
+})
+export class ManagersComponent implements OnInit, OnDestroy {
+  private unsubscribe: Subject<void> = new Subject();
+  public visibleFields = false;
+  public manager: Manager[];
+  public managerId;
+  managerForm = new FormGroup({
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/managers.component.ts]: add visibility modifiers to all the properties and methods (forms and life-cycle hooks also conts)

File: src/app/admin-users/managers/managers.component.ts [hunk 24]
```diff
@@ -0,0 +1,61 @@
...
+    name: new FormControl(''),
+    surname: new FormControl(''),
+    birthday: new FormControl(''),
+    childrenHouseID: new FormControl(''),
+    patronymic: new FormControl(''),
+    photoPath: new FormControl(''),
+    rating: new FormControl(''),
+  });
+  constructor(private managerService: ManagersService, private route: ActivatedRoute) { }
```
Comment (ksenya-ta) [inline src/app/admin-users/managers/managers.component.ts]: you can keep 1 empty line between the methods, it helps to read it easily , no need to remove all of it

**Comment by ksenya-ta** [comment]
all not fixed issues will be fixed in scope of additional tech debt tasks in next sprint

**Pull request merged**